Civi/Core/Paths.php

```diff
@@ -55,6 +56,24 @@ public function __construct() {
       ->register('civicrm.files', function () {
         return \CRM_Core_Config::singleton()->userSystem->getDefaultFileStorage();
       })
+      ->register('wp.frontend.base', function () {
+        return array('url' => CIVICRM_UF_BASEURL);
+      })
+      ->register('wp.frontend', function () use ($paths) {
+        $config = \CRM_Core_Config::singleton();
+        $suffix = defined('CIVICRM_UF_WP_BASEPAGE') ? CIVICRM_UF_WP_BASEPAGE : $config->wpBasePage;
+        return array(
+          'url' => $paths->getVariable('wp.frontend.base', 'url') . $suffix,
+        );
+      })
+      ->register('wp.backend.base', function () {
+        return array('url' => CIVICRM_UF_BASEURL . 'wp-admin/');
```

Comment:
@kcristiano Are you sure that `wp-admin/` should be hard-coded here? The [`admin_url`](https://developer.wordpress.org/reference/hooks/admin_url/) filter allows this to be overridden, so if this path has to be correct without bootstrapping WordPress, then perhaps there needs to be a setting for it somewhere? OTOH, perhaps I'm raising an issue that's outside the scope of this PR.

Reply:
@christianwach I see the point and had originally done that.  In one of the iterations (this PR in various forms dates back to Ft Collins Sprint in June 2016), @totten  made the followimng PRs that have been added:  https://github.com/kcristiano/civicrm-core/commit/48c4fe6f4a9c61f1bc701a2f81377adab7f3c4df  and https://github.com/kcristiano/civicrm-core/commit/8b0c250621cf2a433ba2c6c4d121715671b8f024    This section https://github.com/kcristiano/civicrm-core/commit/8b0c250621cf2a433ba2c6c4d121715671b8f024#diff-46f9229e858ccffc107ef60a91809c43L286  removed the call to admin_url().   I attempted to add back, but had errors depending on the configuration.

As I would much rather have this in and then improve than hold it back even longer, I consider switching to admin_url() as a TODO.   We can create an issue for that and fix it after these changes are tested through an RC.   As this is an installer patch and it affects both civicrm-core and civicrm-wordpress we'll get more testing once it hits RC.

Reply:
@kcristiano Gotcha. Seems like a good idea.